core/block.js

```diff
@@ -1300,8 +1317,8 @@ Blockly.Block.prototype.jsonInit = function(json) {
  * @param {boolean=} opt_disableCheck Option flag to disable overwrite checks.
  */
 Blockly.Block.prototype.mixin = function(mixinObj, opt_disableCheck) {
-  if (goog.isDef(opt_disableCheck) && !goog.isBoolean(opt_disableCheck)) {
-    throw new Error("opt_disableCheck must be a boolean if provided");
+  if (opt_disableCheck !== undefined && typeof opt_disableCheck != 'boolean') {
```

Comment:
This expression uses both `!==` and `!=`, is that intentional? Also, why not also use typeof when checking whether something is (or is not) undefined?

Reply:
!= isn't useful for checking against undefined, since it will also catch null, '', and other falsy values.  Whereas comparing against the string 'boolean' does not require !==.

`typeof opt_disableCheck != 'undefined'` is more verbose and less clear and slower than just `opt_disableCheck !== undefined`